stdlib/ipaddr/ipaddr.rbs

```diff
@@ -0,0 +1,247 @@
+# IPAddr provides a set of methods to manipulate an IP address.  Both IPv4 and
+# IPv6 are supported.
+#
+# ## Example
+#
+#     require 'ipaddr'
+#
+#     ipaddr1 = IPAddr.new "3ffe:505:2::1"
+#
+#     p ipaddr1                   #=> #<IPAddr: IPv6:3ffe:0505:0002:0000:0000:0000:0000:0001/ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff>
+#
+#     p ipaddr1.to_s              #=> "3ffe:505:2::1"
+#
+#     ipaddr2 = ipaddr1.mask(48)  #=> #<IPAddr: IPv6:3ffe:0505:0002:0000:0000:0000:0000:0000/ffff:ffff:ffff:0000:0000:0000:0000:0000>
+#
+#     p ipaddr2.to_s              #=> "3ffe:505:2::"
+#
+#     ipaddr3 = IPAddr.new "192.168.2.0/24"
+#
+#     p ipaddr3                   #=> #<IPAddr: IPv4:192.168.2.0/255.255.255.0>
+#
+class IPAddr
+  include Comparable
+
+  # Creates a new ipaddr containing the given network byte ordered string form of
+  # an IP address.
+  #
+  def self.new_ntoh: (untyped addr) -> IPAddr
+
+  # Convert a network byte ordered string form of an IP address into human
+  # readable form.
+  #
+  def self.ntop: (untyped addr) -> String
+
+  # Creates a new ipaddr object either from a human readable IP address
+  # representation in string, or from a packed in_addr value followed by an
+  # address family.
+  #
+  # In the former case, the following are the valid formats that will be
+  # recognized: "address", "address/prefixlen" and "address/mask", where IPv6
+  # address may be enclosed in square brackets (`[' and `]').  If a prefixlen or a
+  # mask is specified, it returns a masked IP address.  Although the address
+  # family is determined automatically from a specified string, you can specify
+  # one explicitly by the optional second argument.
+  #
+  # Otherwise an IP address is generated from a packed in_addr value and an
+  # address family.
+  #
+  # The IPAddr class defines many methods and operators, and some of those, such
+  # as &, |, include? and ==, accept a string, or a packed in_addr value instead
+  # of an IPAddr object.
+  #
+  def initialize: (?String addr, ?untyped family) -> IPAddr
+
+  # Returns a new ipaddr built by bitwise AND.
+  #
+  def &: (untyped other) -> IPAddr
+
+  # Returns a new ipaddr built by bitwise left shift.
+  #
+  def <<: (untyped num) -> IPAddr
+
+  # Compares the ipaddr with another.
+  #
+  def <=>: (untyped other) -> Integer?
+
+  # Returns true if two ipaddrs are equal.
+  #
+  def ==: (untyped other) -> bool
+
+  alias === include?
+
+  # Returns a new ipaddr built by bitwise right-shift.
+  #
+  def >>: (untyped num) -> untyped
+
+  # Checks equality used by Hash.
+  #
+  def eql?: (untyped other) -> untyped
+
+  # Returns the address family of this IP address.
+  #
+  def family: () -> untyped
+
+  # Returns a hash value used by Hash, Set, and Array classes
+  #
+  def hash: () -> untyped
```

Comment:
@connorshea Some methods are left `untyped`. Do you have a plan to give more precise types in this PR?

Reply:
@soutaro I've updated the file to type more of the methods now, there are still some with untyped but either they couldn't be typed (e.g. `==` can be compared with anything) or I couldn't figure out the correct type.